compiler.go

```diff
@@ -518,6 +525,10 @@ func (t *task) link(parseRes parser.Result, deps linker.Files) (linker.File, err
 		}
 		file.PopulateSourceCodeInfo()
 	}
+
+	if !t.e.c.RetainASTs {
+		file.RemoveAST()
+	}
```

Comment:
This is one significant improvement for memory usage in `protocompile`. It allows the AST to be dropped early in the process, so a large compilation doesn't accumulate a large amount of ASTs pinned on the heap.